docs/DeveloperGuide.md

```diff
@@ -2,37 +2,207 @@
+    - Checked with `isBye()`
+  - `execute()`: An abstract method that must be implemented by all subclasses
+    - Defines the primary action performed by the command (E.g. add new recipe to `recipeList` for `AddRecipeCommand`)
+
+Shown below is a class diagram detailing the interaction between the parent `Command` class and it's child classes:
+![commandClasses.png](img/ClassDiagrams/commandsClassDiagram.png)
+
+#### 2. Command Flow (Using `Command` Classes)
+1. Initialisation: Instantiate a `Command` subclass by passing user input through the `parseCommand()` of `Parser`
+2. Exit signal: If a command is meant to terminate the app (E.g. `ByeCommand`), set `isBye` to true
+3. Execution: Call `execute()`, performing the desired action and interacting with YMFC's data and UI components
+
+
+#### 3. Adding new `Command` Classes
+- Create a `Command` class:
+  1. Inherit from `Command`: Each new command class extends `Command`, inheriting its structure and providing a specific 
+implementation of the `execute()` method
+  2. Override `execute()`: Implement `execute()` to define how the command should interact with `RecipeList`,
+`IngredientList`, `Ui`, and `Storage`
+  3. Additional methods (Optional): You may add helper methods within the class if necessary
+- Register the command
+  1. Update the `parseCommand()` method in `Parser` to recognise and instantiate the new command
+
+#### Example of `Command` Child Classes
+
+1. `AddRecipeCommand` Class
+The AddRecipeCommand class adds a newly created Recipe object to the currentRecipeList object's ArrayList of Recipes.
+
+Shown below is a sequence diagram detailing how a new recipe is added using the AddRecipeCommand object,
+starting from parsing the user input to the addition of the recipe to the recipeList object,
+to finally printing the "recipe added" message to the User on the CLI:
+![AddRecipes.png](img/SequenceDiagrams/AddRecipesSequenceDiagram.png)
+
+2. `EditCommand` Class
+The EditCommand class finds an existing recipe in recipelist by name, and then replaces its parameter
+with the new parameters that the user inputted.
+
+Shown below is a sequence diagram detailing how an existing recipe is modified using the EditCommand object,
+starting from parsing the user input to finding the exiting recipe in the RecipeList object,
+to finally modifying and saving the recipe in the RecipeList:
+![EditCommand.png](img/SequenceDiagrams/EditRecipesSequenceDiagram.png)
+
+### Parser Class
+Self-explanatory, made for parsing user's input command. This class only consist of one public static method 
+`parseCommand()` in order to process input commands.
+The remaining private methods represent separated cases for different commands.
+
+Shown below is a sequence diagram detailing how the Parser class interprets the user's CLI input
+and returns the appropriate command object containing the user's input parameters.
+![Parser.parseCommand.png](img/SequenceDiagrams/Parser.parseCommand.png)
```

Comment:
Looks good but could consider splitting the Parser class to a smaller component if its sequence diagram is so big